compiler/src/dmd/parse.d

```diff
@@ -8249,8 +8249,8 @@ class Parser(AST, Lexer = dmd.lexer.Lexer) : Lexer
                         const len2 = token.len;
                         len = len1 + len2;
                         auto s2 = cast(char*)mem.xmalloc_noscan(len * char.sizeof);
-                        memcpy(s2, s, len1 * char.sizeof);
-                        memcpy(s2 + len1, token.ustring, len2 * char.sizeof);
+                        s2[0 .. len1] = s[0 .. len1]; // Copy len1 elements from s to s2
+                        s2[len1 .. len1 + len2] = token.ustring[0 .. len2]; // Copy len2 elements from token.ustring to s2 starting at len1
```

Comment:
```suggestion
                        s2[0 .. len1] = s[0 .. len1];
                        s2[len1 .. len1 + len2] = token.ustring[0 .. len2];starting at len1
```

Also, please don't add comments that say *what* the code does. Comments should explain *why* the code does something when it's not obvious.